### ISSUE TITLE
Cannot run in inference mode if connected to a python process

### ISSUE THREAD

**Issue by BlueFisher** [Unity-Technologies/ml-agents#3053]
I had upgraded to v0.12.0, but i found that the inference mode would not active if Unity Editor or executor was connected to a python process. The inference mode will activate only when there is a NN model now. It is really not convenient for developers to test their new trained model without an exported NN model.

**Comment by chriselion** [comment]
Hi @BlueFisher,
In 0.12.0, you can specify that Agents run inference even if there's a python process attached by changing the Behavior Type in the Behavior Parameters configuration to "Inference Only"

![image](https://user-images.githubusercontent.com/6877802/70472004-34608480-1a83-11ea-96a2-098e1f3afa05.png)

**Comment by BlueFisher** [comment]
Hi @chriselion ,
If the Behavior Type is "Inference Only" and there is no NN Model, the Unity Editor will throw
```
MissingReferenceException: The variable m_Model of BehaviorParameters doesn't exist anymore.
You probably need to reassign the m_Model variable of the 'BehaviorParameters' script in the inspector.
```
which means the external python process cannot control the agents in inference mode.

BTW, in previous versions, if I set the `train_mode` in `env.reset()` (e.g. `env.reset(train_mode=False)`), both agents in Unity Editor and executor would automatically run in inference mode. But now this configuration seems not working.

**Comment by chriselion** [comment]
Sorry for the delay, this slipped through the cracks over the holidays.

So you want the inference to be performed by python? This should be possible in the latest version - if you're using `mlagents-learn` just omit the `--train` parameter from the commandline.

**Comment by BlueFisher** [comment]
I just find the problem has been solved in [#2990 ](https://github.com/Unity-Technologies/ml-agents/pull/2990)
Because the logic 
```
// If a communicator is enabled/provided, then we assume we are in
// training mode. In the absence of a communicator, we assume we are
// in inference mode.
```
has been remove.
